File: randomForest.R
# Random Forest Algorithm
install.packages('caTools')
library(caTools)


# Import the Dataset
dataset = read.csv('Social_Network_Ads.csv')
dataset = dataset[3:5]

# Encoding the dependent variable (Needed in some classifications)
dataset$Purchased = factor(dataset$Purchased , levels = c(0,1))

# Split the data into train and test sets
set.seed(123)
split = sample.split(dataset$Purchased, SplitRatio = 0.75)
training_set = subset(dataset, split == TRUE)
test_set = subset(dataset, split == FALSE)


# Fitting Random Forest Model
install.packages('randomForest')
library(randomForest)
model = randomForest(x = training_set[-3], y = training_set$Purchased, ntree = 10)


# Predict the test set results
Y_pred = predict(model, newdata = test_set[-3])


# Making the confusion matrix
cm = table(test_set[ ,3], Y_pred)

File: randomForest.py
# -*- coding: utf-8 -*-
"""
Created on Sun Jan  5 18:11:59 2020

@author: 138709
"""

# Random Forest Algorithm

# Import the libraries
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt


# Import the dataset
dataset = pd.read_csv('Social_Network_Ads.csv')
X = dataset.iloc[:,[2,3]].values
Y = dataset.iloc[:,4].values


# Splitting the data into training and test set
from sklearn.model_selection import train_test_split
X_train, X_test, Y_train, Y_test = train_test_split(X, Y, test_size = 0.25, random_state = 0)



# Fitting the randomforest model here
from sklearn.ensemble import RandomForestClassifier
model = RandomForestClassifier(n_estimators = 10, criterion = 'entropy', random_state = 0)
model.fit(X_train, Y_train)


# Predicting the test set results
Y_pred = model.predict(X_test)


# Making the confusion matrix
from sklearn.metrics import confusion_matrix
cm = confusion_matrix(Y_test, Y_pred)
